### DISCUSSION TITLE
用文件夹的方式浏览歌曲文件

### DISCUSSION THREAD

**Discussion by Abcd9183** [navidrome/navidrome#3521] [Ideas]
用文件夹的方式浏览歌曲文件

**Comment by deluan** [comment]
Please, post in English in the future.

Google Translate reply:

https://www.navidrome.org/docs/faq/#-can-you-add-a-browsing-by-folder-optionmode-to-navidrome


虽然从技术上来说可以添加按文件夹浏览选项，但这样做需要对 Navidrome 大部分组件的内部结构进行重大更改。我们已决定专注于符合我们愿景的功能，即打造一个强调标签的音乐服务器。实现文件夹浏览不仅是一项艰巨的任务，而且还可能使支持 Navidrome 的所有当前和未来功能变得更加困难和容易出错。

以下是用户可能认为文件夹浏览很重要的几种情况，以及 Navidrome 计划如何解决这些问题：

按类别（例如流派）对音乐进行分组：Navidrome 已经处理了流派，您可以在 Subsonic 客户端中按流派浏览，将来还会有专门的流派视图。它还将支持多值grouping标签，并具有专门的视图。
同一张专辑有不同的版本：此功能将很快得到[支持](https://github.com/navidrome/navidrome/pull/2162) 。
未标记其库的用户：我们明确不支持这种做法，因为这会使支持 Navidrome 已有和将要拥有的所有功能变得非常困难。我们不希望有代码“推断”包含一堆 MP3 文件的文件夹是一张专辑，因为这种方法会使代码变得非常复杂且容易出错。
如果按文件夹浏览对您来说是一项必不可少的功能，那么还有[其他音乐服务器](https://github.com/basings/selfhosted-music-overview) 提供此功能。如果文件夹浏览是您优先考虑的事情，我们鼓励您探索这些选项。
